src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2381,21 +2381,21 @@ private bool ConsumeCommentMultiSegment()
             byte marker = localBuffer[0];
             if (marker != JsonConstants.Slash && marker != JsonConstants.Asterisk)
             {
-                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.InvalidCharacterAtStartOfComment, marker);
+                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWhileReadingComment, marker);
```

Comment:
Use `InvalidCharacterAtStartOfComment` here. `UnexpectedEndOfDataWhileReadingComment` doesn't make sense for this case.